idea/src/org/jetbrains/kotlin/idea/inspections/CanBePrimaryConstructorPropertyInspection.kt

```diff
@@ -58,4 +65,16 @@ class CanBePrimaryConstructorPropertyInspection : AbstractKotlinInspection() {
             )
         })
     }
+
+    private fun KtClass.isOpen(): Boolean {
+        return hasModifier(KtTokens.OPEN_KEYWORD) || hasModifier(KtTokens.ABSTRACT_KEYWORD) || hasModifier(KtTokens.SEALED_KEYWORD)
+    }
+
+    private fun KtParameter.isUsedInClassInitializer(containingClass: KtClass): Boolean {
+        val name = this.name
+        val classInitializer = containingClass.body?.declarations?.firstIsInstanceOrNull<KtClassInitializer>() ?: return false
+        return classInitializer.anyDescendantOfType<KtNameReferenceExpression> {
+            it.text == name && it.mainReference.resolve() == this
+        }
```

Comment:
Please reuse the function `hasUsages`. 
```suggestion
        val classInitializer = containingClass.body?.declarations?.firstIsInstanceOrNull<KtClassInitializer>() ?: return false
        return hasUsages(classInitializer)
```